.github/workflows/aws-deploy.yml

```diff
@@ -1,48 +1,62 @@
 ---
-    name: AWS Deployment
-    
-    on: [push]
-
-           
-    env:
-      
-      AWS_REGION: ${{ vars.AWS_DEFAULT_REGION }}
-      # AWS resources are spread across different AWS accounts: aics-sw and public_data_releases
-      # The CloudFront distributions is in the aics-sw account
-      # The S3 buckets are in the public_data_releases account
-      CLOUDFRONT_AWS_ACCOUNT_ID: ${{ vars.AWS_AICS_SW_ACCOUNT_ID }}
-      S3_AWS_ACCOUNT_ID: ${{ vars.AWS_PUBLIC_DATA_RELEASES_ACCOUNT_ID }}
-      STAGING_CLOUDFRONT_DISTRIBUTION_ID: ${{ secrets.STAGING_CLOUDFRONT_DISTRIBUTION_ID }}
-      STAGING_S3_BUCKET: s3://staging-ipub-simularium
-      PRODUCTION_CLOUDFRONT_DISTRIBUTION_ID: ${{ secrets.PRODUCTION_CLOUDFRONT_DISTRIBUTION_ID }}
-      PRODUCTION_S3_BUCKET: s3://production-ipub-simularium
-    
-    permissions:
-        id-token: write  # Required for requesting the JWT and OIDC
-        contents: write  # Required for actions/checkout and OIDC tokens
-    
-    jobs:
-      deploy-staging:
-          if: github.ref == 'refs/heads/main'
-          runs-on: ubuntu-latest
-
-          steps:
+name: AWS Deployment
+
+on: [push]
+
+env:
+    AWS_REGION: ${{ vars.AWS_DEFAULT_REGION }}
+    # AWS resources are spread across different AWS accounts: aics-sw and public_data_releases
+    # The CloudFront distributions is in the aics-sw account
+    # The S3 buckets are in the public_data_releases account
+    CLOUDFRONT_AWS_ACCOUNT_ID: ${{ vars.AWS_AICS_SW_ACCOUNT_ID }}
+    S3_AWS_ACCOUNT_ID: ${{ vars.AWS_PUBLIC_DATA_RELEASES_ACCOUNT_ID }}
+    STAGING_CLOUDFRONT_DISTRIBUTION_ID: ${{ secrets.STAGING_CLOUDFRONT_DISTRIBUTION_ID }}
+    STAGING_S3_BUCKET: s3://staging-ipub-simularium
+    PRODUCTION_CLOUDFRONT_DISTRIBUTION_ID: ${{ secrets.PRODUCTION_CLOUDFRONT_DISTRIBUTION_ID }}
+    PRODUCTION_S3_BUCKET: s3://production-ipub-simularium
+
+permissions:
+    id-token: write # Required for requesting the JWT and OIDC
+    contents: write # Required for actions/checkout and OIDC tokens
+
+jobs:
+    check-build:
+        if: github.ref != 'refs/heads/main'
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout
+              uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744
+              # Compute a short sha for use in the OIDC session name, which has a 64 character limit
+            - name: Add SHORT_SHA env property with commit short sha
+              run: echo "SHORT_SHA=`echo ${{ github.sha }} | cut -c1-8`" >> $GITHUB_ENV
+            - name: Setup Bun
+              uses: oven-sh/setup-bun@8f24390df009a496891208e5e36b8a1de1f45135
+            # until vite fixes it's webworker issues the dev build needs a local viewer, but the prod build
+            # uses the viewer from the npm package
+            - run: bun install-viewer
+            - run: bun install
+            - run: bun run build
```

Comment:
We have a workaround in TFE that checks for the current build environment and changes the type of worker, but since it's with the `workerpool` library I'm not sure if it's applicable here.

https://github.com/allen-cell-animated/timelapse-colorizer/blob/main/src/colorizer/loaders/UrlArrayLoader.ts#L49-L57

Reply:
ok I'll try that out thanks!